src/libraries/System.Threading.Tasks.Parallel/src/System.Threading.Tasks.Parallel.csproj

```diff
@@ -7,7 +7,7 @@
   </PropertyGroup>
   <PropertyGroup>
     <TargetPlatformIdentifier>$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)'))</TargetPlatformIdentifier>
-    <FeatureWasmThreads Condition="'$(TargetPlatformIdentifier)' == 'browser' and '$(MonoWasmBuildVariant)' == 'multithread'">true</FeatureWasmThreads>
+    <FeatureWasmThreads Condition="'$(TargetOS)' == 'browser' and '$(MonoWasmBuildVariant)' == 'multithread'">true</FeatureWasmThreads>
```

Comment:
what is the old thing, `TargetPlatformIdentifier`?

Reply:
Why not do it consistently in all project files if `TargetPlatformIdentifier` doesn't work well ?

Reply:
it is the platform identifier from $(TargetFramework). I think it would have the value 'browser' if the target framework was something like `net9.0-browser`

Reply:
That's interesting. `TargetFramework` is not in the form of `net9.0-browser`. We have quite a few places where we use the same condition, not connected with threading. It should mean that they are not triggered, e.g.
https://github.com/dotnet/runtime/blob/135fec006e727a31763271984cd712f1659ccbd3/src/libraries/System.Net.WebSockets/src/System.Net.WebSockets.csproj#L51
https://github.com/dotnet/runtime/blob/135fec006e727a31763271984cd712f1659ccbd3/src/libraries/System.Runtime.InteropServices.JavaScript/src/System.Runtime.InteropServices.JavaScript.csproj#L19

Reply:
@radekdoulik i didn't understand why/how you resolved this. We have many same conditions in the other project files. Do they all work or not ? Could we have them consistent ?

Reply:
> @radekdoulik i didn't understand why/how you resolved this. We have many same conditions in the other project files. Do they all work or not ? Could we have them consistent ?

it depends on the target frameworks of the project. if there are multiple, including the [targetframework]-browser then if works for that framework and it is ok to use it in the condition.

for example System.Runtime.InteropServices.JavaScript.csproj has

`<TargetFrameworks>$(NetCoreAppCurrent)-browser;$(NetCoreAppCurrent)</TargetFrameworks>`

so in this case we want to have the condition use the `TargetPlatformIdentifier`.

in this PR's case we have just single target framework

`<TargetFramework>$(NetCoreAppCurrent)</TargetFramework>`

and so the `TargetPlatformIdentifier` is empty. thus I modified the condition to use `TargetOS` instead

Reply:
could we use `TargetOS` in all places ?

Reply:
I guess it depends on the reasons which led to having multiple target frameworks

Reply:
generally I think we don't want wasm specific code in the non-browser targets, so I think we should differentiate

Reply:
We discourage using `TargetOS` to vary builds (except for tests) because it defaults to the host OS if a target OS is not explicitly passed in, which is the case if you build a library from VS or in the "allConfiguration" CI build.

The correct solution here is to add `$(NetCoreAppCurrent)-browser` TFM and use `TargetPlatformIdentifier` in the condition.

Reply:
@akoeplinger, @lewing I am curious, how are the produced assemblies used further in the process. So in this case, where does the assembly built with net9.0 TFM ends up and where the net9.0-browser one. I imagine they end up in the right Sdk and workload so that users projects get the right assembly?

Reply:
I see many `<IgnoreForCI Condition="'$(TargetOS)' == 'browser'">true</IgnoreForCI>` but I don't know what `IgnoreForCI` does.

There are also many `<DebuggerSupport Condition="'$(DebuggerSupport)' == '' and '$(TargetOS)' == 'browser'">true</DebuggerSupport>`

Overall we have `'$(TargetOS)' != 'browser'` 15x and `'$(TargetOS)' == 'browser'` 126x

Should we fix it all ?

Reply:
And just to be clear: the current way "works" because in CI/official builds we'll always have `TargetOS=browser`, the correct way is mostly important for VS scenarios e.g. you should be able to use the TFM dropdown in VS to select a different config and hit build to build the "browser' version, this likely won't work today anyway because of MonoWasmBuildVariant.